src/deadline_worker_agent/sessions/session.py

```
@@ -1069,14 +1076,30 @@ def update_action(self, action_status: ActionStatus) -> None:
         ):
             self._action_updated_impl(action_status=action_status, now=now)
 
-    def _action_output_log_filter_callback(
+    def action_output_log_filter_callback(
         self, message_type: ActionOutputMessageKind, value: Any
     ) -> None:
         """Callback for the action output log filter
         This callback is called when the output log filter is triggered.
         """
         if message_type == ActionOutputMessageKind.JA_SNAPSHOT:
             self.add_manifest_path(root=value["root"], path=value["manifest"])
+        if (
+            MANIFEST_REPORTING_FEATURE
+            and message_type == ActionOutputMessageKind.JA_UPLOAD
+        ):
+            try:
+                manifest_list_data = json.loads(value)
+                self._upload_manifest_list = [UploadManifestInfo(**item) for item in manifest_list_data]
+            except (TypeError, ValueError) as e:
+                logger.error(
+                    f"Failed to parse manifest snapshot: {e}",
+                    exc_info=True,
+                )
+                self._upload_manifest_list = []
```

Does this mean we would fail silently?